app/validation/validator.py

```diff
@@ -559,18 +568,30 @@ def _get_numeric_range_values(self, answer, answer_ranges):
         }
 
     def validate_duplicates(self, json_to_validate):
-        special_keys = ['id']
+        """
+        question_id & answer_id should be globally unique with some exceptions:
+            - within a block, ids can be duplicated across variants, but must still be unique outside of the block.
+        """
+
         duplicate_errors = []
 
-        for special_key in special_keys:
-            unique_items = []
+        items_per_block = defaultdict(set)
```

Comment:
I think this would be clearer if we named it something like `unique_ids_per_block`

Reply:
Fixed